pkg/provider/azure.go

```diff
@@ -133,6 +133,10 @@ type Config struct {
 	ResourceGroup string `json:"resourceGroup,omitempty" yaml:"resourceGroup,omitempty"`
 	// The location of the resource group that the cluster is deployed in
 	Location string `json:"location,omitempty" yaml:"location,omitempty"`
+	// The name of site where the cluster will be deployed to that is more granular than the region sepecified by the "location" field.
+	ExtendedLocationName string `json:"extendedLocationName,omitempty" yaml:"extendedLocationName,omitempty"`
+	// The type of site that is being targeted.
+	ExtendedLocationType string `json:"extendedLocationType,omitempty" yaml:"extendedLocationType,omitempty"`
```

Comment:
Could you add comments that these two new configs are only applied to LoadBalancer and PublicIP?

Reply:
They will be applied to other things as well mainly persistent volumes

Reply:
thanks for confirming and updated comments